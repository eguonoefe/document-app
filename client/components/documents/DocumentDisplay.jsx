import React from 'react';
import PropTypes from 'prop-types';
import TinyMCE from 'react-tinymce';

import InputField from '../common/InputField';
import Dropdown from '../common/Dropdown';
/**
 * Document Display component for creating a document
 *
 * @param {Object} props { onChange, onSubmit, document, getContent }
 * @returns {String} HTML markup for the Document Display
 */
const DocumentDisplay = ({ onChange, onSubmit, document, getContent }) => (
  <div className="document-form">
    <div className="center hover">
      <h5>Create Document</h5>
      <InputField
        name="title"
        value={document.title}
        placeholder="Document Title"
        className="validate div-design"
        type="text" onChange={onChange} />
      <select
        name="access"
        className="browser-default input-field select" onChange={onChange}>
        <Dropdown value="" text="Select access type" />
        <Dropdown value="public" text="Public" />
        <Dropdown value="private" text="Private" />
        <Dropdown value="role" text="Role" />
      </select>
      <TinyMCE
        content={document.content}
        config={{
          plugins: 'link image code',
          height: 200,
          toolbar:
          'undo redo | bold italic | alignleft aligncenter alignright | code'
        }}
        onChange={getContent}
      />
      <div className="input-field center">
        <button
          className="waves-effect btn button-design" onClick={onSubmit}
          id="save-document">
          Create
          </button>
      </div>
    </div>
  </div>
);

DocumentDisplay.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  document: PropTypes.object.isRequired,
  getContent: PropTypes.func.isRequired
};

export default DocumentDisplay;
