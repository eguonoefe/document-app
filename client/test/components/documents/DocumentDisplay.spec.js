/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import DocumentDisplay from '../../../components/documents/DocumentDisplay';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/documents/DocumentDisplay');

describe('DocumentDisplay Component', () => {
  test('should match the document Display snapshot', () => {
    const document = {
      title: '',
      content: '',
      access: ''
    };
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const getContent = jest.fn();
    const component = shallow(
      <DocumentDisplay
        document={document}
        onChange={onChange}
        onSubmit={onSubmit}
        getContent={getContent} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
