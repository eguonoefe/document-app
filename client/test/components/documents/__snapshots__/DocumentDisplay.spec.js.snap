// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentDisplay Component should match the document Display snapshot 1`] = `
<div
  className="document-form"
>
  <div
    className="center hover"
  >
    <h5>
      Create Document
    </h5>
    <InputField
      className="validate div-design"
      name="title"
      onChange={[MockFunction]}
      placeholder="Document Title"
      type="text"
      value=""
    />
    <select
      className="browser-default input-field select"
      name="access"
      onChange={[MockFunction]}
    >
      <Dropdown
        text="Select access type"
        value=""
      />
      <Dropdown
        text="Public"
        value="public"
      />
      <Dropdown
        text="Private"
        value="private"
      />
      <Dropdown
        text="Role"
        value="role"
      />
    </select>
    <TinyMCE
      config={
        Object {
          "height": 200,
          "plugins": "link image code",
          "toolbar": "undo redo | bold italic | alignleft aligncenter alignright | code",
        }
      }
      content=""
      onChange={[MockFunction]}
    />
    <div
      className="input-field center"
    >
      <button
        className="waves-effect btn button-design"
        id="save-document"
        onClick={[MockFunction]}
      >
        Create
      </button>
    </div>
  </div>
</div>
`;
